### PR TITLE
Fix wikimg

### PR DESCRIPTION
**Pull request by duck-rh**
Changes proposed in this pull request:

- fix old wiki links (various problems)

I confirm that this pull request was submitted according to the [contribution guidelines](/CONTRIBUTING.md): @duck-rh

### CODE REVIEW

**Comment by duck-rh** [comment]
Btw, the BIDI problem was not related to the old wiki stuff, but was in the way of my regex, so…
I still wonder how we got this in.

File: source/community/events/archives/workshop/workshop-march-2012.html.md [hunk 1]
```diff
@@ -31,7 +31,7 @@ wiki_last_updated: 2012-03-22
 
 ### Slides
 
-[Media:Workshop-1203-vdsm-mom.odp‎](Media:Workshop-1203-vdsm-mom.odp‎)
+[Media:Workshop-1203-vdsm-mom.odp](Media:Workshop-1203-vdsm-mom.odp)
```
Comment (garrett) [inline source/community/events/archives/workshop/workshop-march-2012.html.md]: This is some weird MediaWiki-generated fake link. It won't work and needs further rewriting to do so.

File: source/community/get-involved/resources/slide-decks.html.md [hunk 2]
```diff
@@ -68,13 +68,13 @@ This is a large presentation, with notes. It can be used as is, or cut to fit a
 
 *   oVirt Overview:[OVirt-Overview.odp](/images/8/84/OVirt-Overview.odp) (ODP)
 *   oVirt Community Status:[Ovirt_Community_Update-2014-FISL.odp](/images/1/15/Ovirt_Community_Update-2014-FISL.odp) (ODP)
-*   oVirt Installation Process:[OVirt-Installation.odp‎](/images/3/36/OVirt-Installation.odp) (ODP)
-*   oVirt 3.4 New Features:[OVirt34-NewFeatures.odp‎](/images/3/34/OVirt34-NewFeatures.odp) (ODP)
+*   oVirt Installation Process:[OVirt-Installation.odp](/images/3/36/OVirt-Installation.odp) (ODP)
+*   oVirt 3.4 New Features:[OVirt34-NewFeatures.odp](/images/3/34/OVirt34-NewFeatures.odp) (ODP)
```
Comment (garrett) [inline source/community/get-involved/resources/slide-decks.html.md]: These two are still weird and won't work without further changes.

File: source/community/get-involved/resources/slide-decks.html.md [hunk 3]
```diff
@@ -68,13 +68,13 @@ This is a large presentation, with notes. It can be used as is, or cut to fit a
...
 *   V2V - Migrating virtual machines to oVirt:[Virt-v2v.odp](/images/3/30/Virt-v2v.odp) (ODP)
 
 ## FISL 2014 Presentations
 
-*   Elastic Overkill: Is Cloud Really The Be-All End-All for Everyone?:[FISL15_ElasticOverkill.odp‎](/images/b/bf/FISL15_ElasticOverkill.odp) (ODP)
+*   Elastic Overkill: Is Cloud Really The Be-All End-All for Everyone?:[FISL15_ElasticOverkill.odp](/images/b/bf/FISL15_ElasticOverkill.odp) (ODP)
```
Comment (garrett) [inline source/community/get-involved/resources/slide-decks.html.md]: Anything with /images/?/??/* won't work unless it's rewritten.

File: source/community/get-involved/resources/slide-decks.html.md [hunk 4]
```diff
@@ -95,11 +95,11 @@ Arch & Work in progress: [(PDF)](/w/images/b/be/Ovirt-arch-iheim.pdf)
 
 oVirt Node: [(PDF)](/wp-content/uploads/2011/11/ovirt-node.pdf)
 
-oVirt API: ![](OVirt-API-CLI-SDK-20111102.pdf  "fig:OVirt-API-CLI-SDK-20111102.pdf ") (PDF)
+oVirt API: [(PDF)](/images/wiki/OVirt-API-CLI-SDK-20111102.pdf)
```
Comment (garrett) [inline source/community/get-involved/resources/slide-decks.html.md]: Much better, but odd that there's [()] — perhaps we should do another pass and remove the () from links? (Stylistic, not functional suggestion.)

Same goes for any below that also have the odd [()] construct.

File: source/develop/api/design/vdsm-api.html.md [hunk 5]
```diff
@@ -31,7 +31,7 @@ Key design points:
 *   No business logic will be implemented in the QMF agent or REST API
 *   Both QMF and REST are optional components (ie. they are not required for vdsm to function correctly). However at least one of these is needed to enable management.
 
-![](Vdsm-api-architecture.png‎ "Vdsm-api-architecture.png‎")
+![](Vdsm-api-architecture.png "Vdsm-api-architecture.png")
```
Comment (garrett) [inline source/develop/api/design/vdsm-api.html.md]: This still has the extra redundant junk at the end of the link. (That is: Everything in the parentheses that are basically the space and whatever is in the quotes, including the quotes themselves.)

File: source/develop/release-management/features/engine/separate-reports-host.html.md [hunk 6]
```diff
@@ -436,7 +436,7 @@ Some installations might want to separate for security reasons, e.g. to give som
 
 ## Presentation
 
-![](Engine-dwh-reports-on-separate-hosts-presentation.zip "fig:Engine-dwh-reports-on-separate-hosts-presentation.zip") pandoc source and the generated html included.
+[ZIP](/images/wiki/Engine-dwh-reports-on-separate-hosts-presentation.zip) pandoc source and the generated html included.
```
Comment (garrett) [inline source/develop/release-management/features/engine/separate-reports-host.html.md]: Note: I still think it's wrong and odd to have non-image content under images. (It should reside elsewhere.)

File: source/develop/release-management/features/infra/taskmanagerdetailed.html.md [hunk 7]
```diff
@@ -127,14 +127,14 @@ The *categories* field will enable filtering jobs by categories. An action type
 ##### Main Task Manager Class Diagram
 
 The following class diagrams describes the entities participating in the the Task Manager feature:
- ![](async-task-main-class-diagram.jpeg "fig:async-task-main-class-diagram.jpeg")
+ ![](/images/wiki/Async-task-main-class-diagram.jpeg)
```
Comment (garrett) [inline source/develop/release-management/features/infra/taskmanagerdetailed.html.md]: This link, and several below, have been changed to have an uppercase first letter. Is that intentional? It doesn't matter as long as this link (and the others similar to it) has the same case as the filename on disk.

Note: Several others already start with an uppercase letter. It's probably a wiki-ism — but it's odd that this is inconsistent.

File: source/develop/release-management/features/network/detailed-host-network-qos.html.md [hunk 8]
```diff
@@ -177,7 +177,7 @@ Since QoS API has been added in oVirt 3.5, it will be possible to expose Host Ne
 
 <b>Slightly out of date - should be similar, but other values will appear in the dialogs. A new category should appear under the DC/QoS subtab.</b>
 
-![ thumb | right](NetworkDialogQos.png  "fig: thumb | right") ![ thumb | right](InterfaceQosOverride.png  "fig: thumb | right")
+![ thumb | right](/images/wiki/NetworkDialogQos.png) ![ thumb | right](/images/wiki/InterfaceQosOverride.png)
```
Comment (garrett) [inline source/develop/release-management/features/network/detailed-host-network-qos.html.md]: I think the ` thumb | right` isn't correct here. That's supposed to be for alt text, not directives. I'd suggest `Quality of Service network dialog screenshot` and `Quality of Service Override screenshot` as the alt text instead.

File: source/develop/release-management/features/network/detailed-osn-integration.html.md [hunk 9]
```diff
@@ -183,7 +183,7 @@ The integration of network providers into oVirt will be incremental. The followi
 
 ### User Experience
 
-![ thumb | right](Providers.png  "fig: thumb | right") ![ thumb | right](Discovery.png  "fig: thumb | right") ![ thumb | right](Add.png  "fig: thumb | right") ![ thumb | right](Networks.png  "fig: thumb | right")
+![ thumb | right](/images/wiki/Providers.png) ![ thumb | right](/images/wiki/Discovery.png) ![ thumb | right](/images/wiki/Add.png) ![ thumb | right](/images/wiki/Networks.png)
```
Comment (garrett) [inline source/develop/release-management/features/network/detailed-osn-integration.html.md]: More ` thumb | right` issues.

To script it, I guess it would be possible to take the filename, de-camel-case it and add " screenshot" afterward. That would probably be a decent start?

File: source/develop/release-management/features/virt/clone-vm.html.md [hunk 10]
```diff
@@ -45,7 +45,7 @@ As shown in the following picture a new action button called "Clone VM" has been
 
 When clicked, a new window will be opened containing only the new name of the VM:
 
-![](CloneVmWindow.png‎ "CloneVmWindow.png‎")
+![](CloneVmWindow.png "CloneVmWindow.png")
```
Comment (garrett) [inline source/develop/release-management/features/virt/clone-vm.html.md]: This one still has the in-quotes duplication.

File: source/develop/release-management/releases/3.1/featurelist.html.md [hunk 11]
```diff
@@ -11,8 +11,8 @@ wiki_last_updated: 2012-03-22
...
-| 0%         | Task management                                                                    | [ Task management ](Features/TaskManagerDetailed‎)                      | Improving task management.                                       |                                             | 2011-12-01 |
-| 0%         | Locking mechanism                                                                  | [ Locking mechanism ](Features/DetailedLockMechanism‎)                  | A new locking mechanism.                                         |                                             | 2011-12-01 |
+| 0%         | Task management                                                                    | [ Task management ](Features/TaskManagerDetailed)                      | Improving task management.                                       |                                             | 2011-12-01 |
```
Comment (garrett) [inline source/develop/release-management/releases/3.1/featurelist.html.md]: It's not really related, and I'm not sure why this is showing as a change (I guess whitespace somehow), but the links shouldn't have spaces around the text. It isn't a functional issue, but a style one, and should be addressed — but not in the batch of changes.

File: source/documentation/admin-guide/virt/live-snapshots.html.md [hunk 12]
```diff
@@ -26,7 +26,7 @@ An example of a thin provisioned image:
 
 An example of a snapshots:
 
-![](LiveSnapshot_SnapshotImage.png‎ "LiveSnapshot_SnapshotImage.png‎")
+![](LiveSnapshot_SnapshotImage.png "LiveSnapshot_SnapshotImage.png")
```
Comment (garrett) [inline source/documentation/admin-guide/virt/live-snapshots.html.md]: This link still has the redundancy-in-quotes issue... and no prefix for the image either.

**Comment by garrett** [comment]
A _lot_ of those JPEGs should really be PNGs. They should be cleaned up and converted, with their filenames rewritten.

(I did this for the conversion — it's possible that there are duplicates now in this PR.)

**Comment by garrett** [comment]
After doing a basic conversion (without optimization, just for checking) all the jpeg/jpg/JPG files to PNG, I have discovered most have already previously been converted and optimized. 

However, there are a few that haven't been. Here's the list:

	23lten7kbgoqscx67spcyfyy7ce8gcc.png
	DomainRefreshLun.png
	Donnydavis.png
	File-oVirtFOSDEM.png
	Gluster.png
	Gluster2.png
	Gluster3.png
	Gluster4.png
	Libvdsm.png
	Policies-Doron.png
	Tsm-int.png
	external_network_provider_schema1.png

I'll work on these and delete the jpg/jpeg/JPG files in a patch. (Of course, we'll need to change the filenames in the source too, at the same time.)

**Comment by garrett** [comment]
Some of the JPEGs might actually need to be JPEGs (and could be JPEGs in master even). I'm going through each one.

Note:
- 23lten7kbgoqscx67spcyfyy7ce8gcc.png is a meme using copyrighted movie material. It shouldn't belong in the source for this reason as well as being tacky.
- Donnydavis.png should be a JPEG. I've restored it as such.
- File-oVirtFOSDEM.png should also be a JPEG. I've also restored it. I don't see it referenced in any of the Markdown files, however.

The rest have been enhanced with a neural network (to remove JPEG artifacts, yet keep the actual content the same) and optimized to be squashed down to optimal indexed palettes (where appropriate) for file size optimization.

**Comment by garrett** [comment]
There are also a lot of files that aren't using a path (and don't have the right extension). You can see a lot of them in the following grep (from within the `source` dir):

```
grep -R '\][^"]* "[^"]*")' 
```

This doesn't grep for the above mentioned problems, but searches for quoted strings at the end of images and links (which is allowed, but pretty much always misused on the oVirt website). Without limited to files, this does catch some JS files, so use care when rewriting files, and limit to Markdown only (of course).

**Comment by duck-rh** [comment]
I'm on it, but will finish tomorrow, tired.

**Comment by duck-rh** [comment]
More fixes planned with @garrett 

With the following problems I could not fix because the images are missing:
cannot find 'Ovirt-guest-agent-sso-windows2.png' in page 'source/documentation/internal/guest-agent/windows.html.md'
cannot find 'rhev-agent-flow.png' in page 'source/documentation/internal/guest-agent/guest-agent-automatic-login-rhel6.html.md'
cannot find 'Ipv6.jpg' in page 'source/develop/release-management/features/network/ipv6-support.html.md'
cannot find 'oCoFlowCommandWithTasks.png' in page 'source/develop/release-management/features/engine/command-coordinator-flows-and-events.html.md'

**Comment by duck-rh** [comment]
Also this time I tried to favour png over jpe?g if available.

Also it tries harder to find first the original case and fallback to any case mix. This was useful in some case but still prefer the original naming if both are available (because sometimes files from my archives only differ by extension or case but are different).

File: source/community/events/archives/workshop/workshop-march-2012.html.md (hunk 1 above)
Comment (duck-rh) [inline source/community/events/archives/workshop/workshop-march-2012.html.md]: fixes

**Comment by duck-rh** [comment]
The more I dig the more it's wonderful…

So there are also links like "//www.ovirt.org/images/a/ae/OVirt-Foreman-DUS.odp".

**Comment by duck-rh** [comment]
Another problem I cannot solve:
cannot find 'rhevm-3-1.png' in page 'source/develop/developer-guide/db-issues/dbstructure.html.md'

File: source/community/get-involved/resources/slide-decks.html.md (hunk 2 above)
Comment (duck-rh) [inline source/community/get-involved/resources/slide-decks.html.md]: fixed

File: source/community/get-involved/resources/slide-decks.html.md (hunk 3 above)
Comment (duck-rh) [inline source/community/get-involved/resources/slide-decks.html.md]: fixed

File: source/develop/api/design/vdsm-api.html.md (hunk 5 above)
Comment (duck-rh) [inline source/develop/api/design/vdsm-api.html.md]: often it is junk but sometimes not, so I cannot automate this without destroying content

File: source/develop/release-management/features/network/detailed-host-network-qos.html.md (hunk 8 above)
Comment (duck-rh) [inline source/develop/release-management/features/network/detailed-host-network-qos.html.md]: This is editorial changes. Unless there is a clear way to automate replacement this is outside the scope of this PR.

File: source/develop/release-management/features/network/detailed-osn-integration.html.md (hunk 9 above)
Comment (duck-rh) [inline source/develop/release-management/features/network/detailed-osn-integration.html.md]: some are screenshots, some slides, some mockups… not sure we can automate this.

File: source/develop/release-management/features/virt/clone-vm.html.md (hunk 10 above)
Comment (duck-rh) [inline source/develop/release-management/features/virt/clone-vm.html.md]: When the quotes contains the filename, then I guess we could automate easily a removal. Probably also some cases with general junk into it. Would need to list cases first. The rest would need human supervision I guess.

File: source/develop/release-management/releases/3.1/featurelist.html.md (hunk 11 above)
Comment (duck-rh) [inline source/develop/release-management/releases/3.1/featurelist.html.md]: I tried to normalized some cases but it was not always easy so probably a lot remaining.

File: source/documentation/admin-guide/virt/live-snapshots.html.md (hunk 12 above)
Comment (duck-rh) [inline source/documentation/admin-guide/virt/live-snapshots.html.md]: fixed (the link)

**Comment by duck-rh** [comment]
@garrett a few "new" old wiki stuff in jpeg format were added in the recent fixes, could you do a final pass before we merge (when ready) please?

**Comment by duck-rh** [comment]
ok, did quotes cleanup, but still ~150 links are non-obvious and may contain useful naming.
